src/sql-parser/src/ast/visit_macro.rs

```diff
@@ -672,7 +672,7 @@ macro_rules! make_visitor {
                     envelope,
                     if_not_exists,
                     materialized,
-                } => visitor.visit_create_source(name, connector, format, envelope, *if_not_exists, *materialized),
+                } => visitor.visit_create_source(name, connector, format.as_auto_ref(), envelope, *if_not_exists, *materialized),
```

Comment:
TIOLI: I think that we can get rid of the `as_auto_ref` methods and just use the now-standard `as_deref` methods on option?

Reply:
I'll leave it as `as_auto_ref` and change all of them to `as_deref` in a separate diff to keep the commit history legit.

Reply:
That's not what `as_auto_ref` is about. This visitor is generic over mutability via a macro. `as_auto_ref` means `as_ref` or `as_mut` depending on whether the macro is currently generating the immutable or mutable visitor.